buildSrc/src/main/java/Dependencies.kt

```diff
@@ -54,7 +54,7 @@ object Versions {
 
     const val google_play_store_version = "1.8.0"
 
-    const val protobuf = "3.11.4" // keep in sync with the version used in AS.
+    const val protobuf = "3.19.1" // keep in sync with the version used in AS.
```

Comment:
Ah, this comment is suspect. I think "AS" stands for App Services (our native components that are used to share code between desktop/iOS/Android) and not Android Studio, so we probably want to have this version in sync: https://github.com/mozilla/application-services/blob/main/build.gradle#L26

@Mugurell can probably confirm if we need to put a PR for app services too.

Reply:
I'm guessing this built locally and worked for you so I wonder what happens when these are longer in sync? We might be able to tell when the UI tests run in this build.

Reply:
Oh, that makes way more sense. I'd been deep in a search trying to figure out what version Studio was using. Happy to put up a PR for app services as well if that's the best path forward. Need to get my tooling together for it still, anyway

Reply:
> 80 test cases failed, 100 passed, 8 flaky

Sounds like we need to upgrade the a-s version too, which may be more complicated at that layer. We can start by filing an with our intent.

```
java.lang.ClassCastException: java.lang.Class cannot be cast to java.lang.String
	at com.google.protobuf.MessageSchema.newSchemaForRawMessageInfo(MessageSchema.java:507)
	at com.google.protobuf.MessageSchema.newSchema(MessageSchema.java:227)
	at com.google.protobuf.ManifestSchemaFactory.newSchema(ManifestSchemaFactory.java:77)
	at com.google.protobuf.ManifestSchemaFactory.createSchema(ManifestSchemaFactory.java:71)
	at com.google.protobuf.Protobuf.schemaFor(Protobuf.java:90)
	at com.google.protobuf.Protobuf.schemaFor(Protobuf.java:104)
	at com.google.protobuf.GeneratedMessageLite.parsePartialFrom(GeneratedMessageLite.java:1537)
	at com.google.protobuf.GeneratedMessageLite.parseFrom(GeneratedMessageLite.java:1711)
	at com.google.protobuf.GeneratedMessageLite.parseFrom(GeneratedMessageLite.java:1704)
	at mozilla.appservices.places.MsgTypes$BookmarkNode.parseFrom(MsgTypes.java:3615)
	at mozilla.appservices.places.PlacesReaderConnection.getBookmarksTree(PlacesConnection.kt:498)
	at mozilla.components.browser.storage.sync.PlacesBookmarksStorage$getTree$2.invokeSuspend(PlacesBookmarksStorage.kt:38)
	at mozilla.components.browser.storage.sync.PlacesBookmarksStorage$getTree$2.invoke(Unknown Source:8)
	at mozilla.components.browser.storage.sync.PlacesBookmarksStorage$getTree$2.invoke(Unknown Source:4)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:165)
```

Reply:
I filed https://github.com/mozilla/application-services/issues/4657.

There's a lot that changed in protobufs since 3.11.4, so it's possible that switchover won't be as easy on the AS side. Good news is that if that does prove to be a problem we can try the workarounds suggested in https://github.com/grpc/grpc-java/issues/7690#issuecomment-747196215 ?

Reply:
> @Mugurell can probably confirm if we need to put a PR for app services too.

When I added this new dependency in Fenix I saw those strange errors posted in the comment below coming from AS incompatibilities with newer versions so I just used what aplication-services uses to ensure everything will continue to work the same as before.
Since the functionalities are different we could probably try forcing one version in Fenix and one in AS but I don't see any reason to go through these hurdles. And no reason to update now in Fenix if the workaround for M1 works with the current version. If I understand correctly AS plans to drop this dependency soon-ish after which just updating in Fenix should be straight-forward.

Reply:
Looks like the UI tests pass using the workaround! Everything looks good locally as far as I can tell